clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -618,7 +618,8 @@ private TransactionManager configureTransactionState(ProducerConfig config,
      * @throws org.apache.kafka.common.errors.UnsupportedVersionException fatal error indicating the broker
      *         does not support transactions (i.e. if its version is lower than 0.11.0.0)
      * @throws org.apache.kafka.common.errors.AuthorizationException fatal error indicating that the configured
-     *         transactional.id is not authorized. See the exception for more details
+     *         transactional.id is not authorized, or the idempotent producer id is unavailable. See the exception for
```

Comment:
What do we mean by idempotent producer ID is unavailable? Is it that the broker is unavailable? And do we return this error if the broker is not available?

Reply:
Should we also change this to not say "fatal error"

Reply:
One could get Authorization issue upon requesting for a producerId, not necessarily caused by an unavailable broker. Yes, maybe remove the word "fatal" thanks!

Reply:
Can we remove fatal here still?

Reply:
🤦 - sorry and done.